sql/src/main/java/io/crate/operation/tablefunctions/GenerateSeriesFunction.java

```diff
@@ -0,0 +1,169 @@
+
+            Long[] values = new Long[arguments.size()];
+            int idx = 0;
+
+            for (Input argument : arguments) {
+                Number value = (Number) argument.value();
+                values[idx++] = value.longValue();
+            }
+
+            return values;
+        }
+
+        @Override
+        public Bucket execute(Collection<? extends Input> arguments) {
+            final Long[] values = extractValues(arguments);
+
+            final long start = values[0];
+            final long stop = values[1];
+            final long step = values.length==3 ? values[2] : 1;
+
+            assert step!=0 : "step must not be 0";
+
+            final int numCols = 1;
+            final int numRows = (int) ((stop - start) / step);
+
+            assert numRows>=0 : "invalid combination: cannot reach "+stop+" from "+start+" by stepping "+step;
+
+            return new Bucket() {
+                final Object[] cells = new Object[numCols];
+                final RowN row = new RowN(cells);
+
+                @Override
+                public int size() {
+                    return numRows+1;
+                }
+
+                @Override
+                public Iterator<Row> iterator() {
+                    return new Iterator<Row>() {
+
+                        int currentRow = 0;
+
+                        @Override
+                        public boolean hasNext() {
+                            return currentRow < size();
+                        }
+
+                        @Override
+                        public Row next() {
+                            if (!hasNext()) {
+                                throw new NoSuchElementException("No more rows");
+                            }
+                            cells[0] = start + currentRow*step;
+                            currentRow++;
+                            return row;
+                        }
+
+                        @Override
+                        public void remove() {
+                            throw new UnsupportedOperationException("remove is not supported for " +
+                                GenerateSeriesFunction.class.getSimpleName() + "$iterator");
+                        }
+                    };
+                }
+            };
+        }
+
+        @Override
+        public TableInfo createTableInfo(ClusterService clusterService) {
+            int noElements = info.ident().argumentTypes().size();
+            Map<ColumnIdent, Reference> columnMap = new LinkedHashMap<>(noElements);
+            Collection<Reference> columns = new ArrayList<>(noElements);
+
+            ColumnIdent columnIdent = new ColumnIdent("generate_series");
+            DataType dataType = DataTypes.LONG;
+            Reference reference = new Reference(
+                new ReferenceIdent(TABLE_IDENT, columnIdent),
+                RowGranularity.DOC, dataType);
+
+            columns.add(reference);
+            columnMap.put(columnIdent, reference);
```

Comment:
It would be possible to do this upfront in the constructor, as the amount of columns that are returned is static and won't change